lib/handleRequest.js

```diff
@@ -23,7 +23,7 @@ function handleRequest (req, res, params, context) {
   if (method === 'POST' || method === 'PUT' || method === 'PATCH') {
     // application/json content type
     if (contentType && contentType.indexOf('application/json') > -1) {
-      return jsonBody(request, reply)
+      return jsonBody(request, reply, context._jsonBodyLimit)
```

Comment:
An object can be better here: it's future proof

Reply:
@allevo Yes I agree it would be future proof. It would be nice to not allocate an object though. It would also be nice to wait until Node 4 support is dropped so we can use object destructuring in `jsonBody`.

I can add it if you insist though.

Reply:
Since this is an internal function we can leave it as is.
It will not be nice, but is fast :)

Reply:
Truly the parser options can be stored inside the route definition aka `Context`. Without create any object at runtime

Reply:
That sounds good to me 👍 
How does this look?

```diff
diff --git a/fastify.js b/fastify.js
index 6d35bdc..d6155ee 100644
--- a/fastify.js
+++ b/fastify.js
@@ -494,7 +494,9 @@ function build (options) {
     this.config = config
     this.errorHandler = errorHandler
     this._middie = middie
-    this._jsonBodyLimit = jsonBodyLimit
+    this._jsonParserOptions = {
+      limit: jsonBodyLimit
+    }
     this._fastify = fastify
   }

diff --git a/lib/handleRequest.js b/lib/handleRequest.js
index d07bd29..79a8bb2 100644
--- a/lib/handleRequest.js
+++ b/lib/handleRequest.js
@@ -23,7 +23,7 @@ function handleRequest (req, res, params, context) {
   if (method === 'POST' || method === 'PUT' || method === 'PATCH') {
     // application/json content type
     if (contentType && contentType.indexOf('application/json') > -1) {
-      return jsonBody(request, reply, context._jsonBodyLimit)
+      return jsonBody(request, reply, context._jsonParserOptions)
     }

     // custom parser for a given content type
@@ -42,7 +42,7 @@ function handleRequest (req, res, params, context) {

     // application/json content type
     if (contentType.indexOf('application/json') > -1) {
-      return jsonBody(request, reply, context._jsonBodyLimit)
+      return jsonBody(request, reply, context._jsonParserOptions)
     }
     // custom parser for a given content type
     if (context.contentTypeParser.fastHasHeader(contentType)) {
@@ -57,7 +57,8 @@ function handleRequest (req, res, params, context) {
   return
 }

-function jsonBody (request, reply, limit) {
+function jsonBody (request, reply, options) {
+  const limit = options.limit
   const contentLength = request.headers['content-length'] === undefined
```

@mcollina @delvedor ?

Reply:
Yes, it’s ok.